File: 08-repeticao-arranjos-e-conjuntos/exemplos/maximo.cpp
#include "bscpp.hpp"
#include <array>

using namespace std;

// Encontra o valor máximo de numeros.
int maximo(array<int, 5> numeros)
{
    int max = numeros[0];
    for (int num : numeros) {
        if (num > max) {
            max = num;
        }
    }
    return max;
}

examples
{
    check_expect(maximo({5, 4, 1, 0, 7}), 7);
    check_expect(maximo({1, -5, 8, 1, 2}), 8);
    check_expect(maximo({-4, -6, -1, -1, -3}), -1);
}

int main()
{
    run_tests();
}

File: 08-repeticao-arranjos-e-conjuntos/exemplos/maximo.py
def maximo(lst: list[int]) -> int:
    '''
    Encontra o valor máximo de *lst*.
    Requer que lst seja não vazia.
    Exemplos
    >>> maximo([7])
    7
    >>> maximo([-1, -4, 0])
    0
    '''
    assert len(lst) != 0
    maximo = lst[0]
    for n in lst:
        if n > maximo:
            maximo = n
    return n
